Fix usage of the primary and secondary organization's colors

andreslucena opened the pull request:
## :tophat: What? Why?

After we merged #10974 , we found lots of places that weren't using the CSS primary/secondary variables colors, and they were using the foundation ones. This PR fixes them.

It also introduces these rules on the spellchecker so we don't make this same mistake in the future. 

Testing for all these issues should be:

1. Change the primary and secondary color
2. Follow the steps for the bug
3. Check also that this change doesn't introduce a bug in the admin panel, as these colors **should not be changed** in the admin panel. I mention this because it's something that I detected on some of these changes and I might have missed a leak somewhere.
4. Notice that these solved bugs are only a selection of the ones that I've found easier to actually see. There are much more but I don't have the energy to actually go to all these changes are they were just too much effort :/ 

## :pushpin: Related Issues
 
- Related to #10974
 
## Solved bugs

### 0. Links 

#### Testing

1. See the links in pretty much all the places of the app

#### :camera: Screenshots

##### Before

![Screenshot of processes list without the secondary color in the links](https://hackmd.io/_uploads/rkLnBlwwn.png)


##### After

![Screenshot of processes list with the secondary color in the links](https://hackmd.io/_uploads/SyVQUlvP2.png)

### 1. Attachments modal progress bar

#### Testing

1. Enable Attachments in a Proposals component
2. Go to create a new Proposal
3. Add an attachment

#### :camera: Screenshots

##### Before 

![Screenshot of upload modal form without the primary color](https://hackmd.io/_uploads/HyplIewvh.png)


##### After


![Screenshot of upload modal form with the primary color](https://hackmd.io/_uploads/H1UIUgwvn.png)

### 2. Autocomplete suggestions

#### Testing

1. Enable the Here.com API or some other autocomplete provider
2. Enable the "Participants can create meetings" in a Meetings component
3. Go to create a new "In person" Meeting
5. Fill the "Address" field

#### :camera: Screenshots

##### Before

![Screenshot of the autocomplete suggestions without the secondary color in the links](https://hackmd.io/_uploads/BJonBxDPh.png)


##### After

![Screenshot of the autocomplete suggestions with the secondary color in the links](https://hackmd.io/_uploads/B1iV8eDv3.png)

### 3. Button shadow

#### Testing

1. Go to a Proposal 
2. See the "Endorse" button

#### :camera: Screenshots

##### Before

![Screenshot of the endorse button without the secondary color in it](https://hackmd.io/_uploads/HJ3-IxwP2.png)

##### After

![Screenshot of the endorse button with the secondary color in it](https://hackmd.io/_uploads/SJUo8evwn.png)


### 4. Assemblies' organization chart

#### Testing

1. Go to /assemblies
2. Scroll to the bottom of the page 

#### :camera: Screenshots

##### Before

![Screenshot of the assemblies' organization chart without the secondary color](https://hackmd.io/_uploads/S1WqE8Dvh.png)


##### After

![Screenshot of the assemblies' organization chart with the secondary color](https://hackmd.io/_uploads/SkVcNIvw3.png)

---

These are just some of the changes introduced by this PR. I actually stopped looking for these places as I've seen things like 

https://github.com/decidim/decidim/blob/9ced3d0f5bc9c90ac85d30c4f184fad20716a562/decidim-core/app/packs/stylesheets/decidim/extras/_impersonation-bar.scss#L6

Ok, a link in the impersonation-bar, lets go see it:

https://github.com/decidim/decidim/blob/develop/decidim-core/app/views/layouts/decidim/_impersonation_warning.html.erb

 There isn't any link!! 😵‍💫

andreslucena committed:
Use CSS variables for upload modal progress bar color

alecslupu commented:
I was checking for `$secondary` and `$primary` through the application and i have found quite a few occurrences in all the source code. Are you aware of this ? Are those CSS definitions subject to change?
Could we do a bigger PR to address this entirely?

```
$grep -RiHn '$secondary' ./decidim*

./decidim-core/app/packs/stylesheets/decidim/_properties.scss:6:  --secondary: #{$secondary};
./decidim-core/app/packs/stylesheets/decidim/_properties.scss:7:  --secondary-rgb: #{$secondary-rgb};
./decidim-core/app/packs/stylesheets/decidim/vizzs/_orgchart.scss:17:    fill: $secondary;
./decidim-core/app/packs/stylesheets/decidim/utils/_settings.scss:73:  secondary: $secondary,
./decidim-core/app/packs/stylesheets/decidim/utils/_settings.scss:203:$anchor-color: $secondary-color;
./decidim-core/app/packs/stylesheets/decidim/utils/_settings.scss:587:$pagination-item-background-current: $secondary-color;
./decidim-core/app/packs/stylesheets/decidim/modules/_conference-media.scss:6:      color: $secondary;
./decidim-core/app/packs/stylesheets/decidim/modules/_buttons.scss:265:    secondary: shade($secondary, 50%),
./decidim-core/app/packs/stylesheets/decidim/modules/_input-tags.scss:18:    color: $secondary;
./decidim-core/app/packs/stylesheets/decidim/modules/_conference-programme.scss:72:    color: $secondary;
./decidim-core/app/packs/stylesheets/decidim/modules/_help.scss:22:  border: 0.25rem solid lighten($secondary, 25%);
./decidim-core/app/packs/stylesheets/decidim/modules/_help.scss:24:  background-color: $secondary;
./decidim-core/app/packs/stylesheets/decidim/modules/_autocomplete.scss:1:$autocomplete-item-color: $secondary !default;
./decidim-core/app/packs/stylesheets/decidim/modules/_inline-filters.scss:23:    color: $secondary;
./decidim-core/app/packs/stylesheets/decidim/modules/_inline-filters.scss:56:        background-color: rgba($secondary, 0.05);
./decidim-core/app/packs/stylesheets/decidim/modules/_input-mentions.scss:68:    color: $secondary;
./decidim-core/app/packs/stylesheets/decidim/modules/_collapsible-list.scss:39:    color: $secondary;
./decidim-core/app/packs/stylesheets/decidim/modules/_cards.scss:136:    color: darken($secondary, 20%);
./decidim-core/app/packs/stylesheets/decidim/modules/_cards.scss:701:    background-color: $secondary;
./decidim-core/app/packs/stylesheets/decidim/modules/_messages.scss:10:  background-color: $secondary;
./decidim-core/app/packs/stylesheets/decidim/_variables.scss:3:$secondary-color: null;
./decidim-core/app/packs/stylesheets/decidim/_variables.scss:14:$secondary: #39747f !default;
./decidim-core/app/packs/stylesheets/decidim/_variables.scss:15:$secondary-rgb: 57, 116, 127 !default;
./decidim-elections/app/packs/stylesheets/decidim/elections/focus/_evote.scss:119:      color: $secondary;
./decidim-elections/app/packs/stylesheets/decidim/elections/focus/_evote.scss:132:      background-color: $secondary;
./decidim-elections/app/packs/stylesheets/decidim/elections/focus/_evote.scss:146:      color: $secondary;
./decidim-elections/app/packs/stylesheets/decidim/elections/focus/_evote.scss:156:      background-color: $secondary;
```

```
$grep -RiHn '$primary' ./decidim*
./decidim-admin/app/packs/stylesheets/decidim/admin/plugins/_foundation-datepicker.scss:4:$calendar-active-color: $primary-color;
./decidim-admin/app/packs/stylesheets/decidim/admin/utils/_settings.scss:202:$anchor-color: $primary-color;
./decidim-admin/app/packs/stylesheets/decidim/admin/utils/_settings.scss:259:$accordion-background: $primary-color;
./decidim-admin/app/packs/stylesheets/decidim/admin/utils/_settings.scss:277:$accordionmenu-arrow-color: $primary-color;
./decidim-admin/app/packs/stylesheets/decidim/admin/utils/_settings.scss:289:$badge-background: $primary-color;
./decidim-admin/app/packs/stylesheets/decidim/admin/utils/_settings.scss:302:$breadcrumbs-item-color: $primary-color;
./decidim-admin/app/packs/stylesheets/decidim/admin/utils/_settings.scss:317:$button-background: $primary-color;
./decidim-admin/app/packs/stylesheets/decidim/admin/utils/_settings.scss:401:$drilldown-arrow-color: $primary-color;
./decidim-admin/app/packs/stylesheets/decidim/admin/utils/_settings.scss:476:$input-outline-focus: 2px solid $primary-color;
./decidim-admin/app/packs/stylesheets/decidim/admin/utils/_settings.scss:490:$label-background: $primary-color;
./decidim-admin/app/packs/stylesheets/decidim/admin/utils/_settings.scss:590:$progress-meter-background: $primary-color;
./decidim-admin/app/packs/stylesheets/decidim/admin/utils/_settings.scss:673:$prototype-separator-background: $primary-color;
./decidim-admin/app/packs/stylesheets/decidim/admin/utils/_settings.scss:750:$slider-handle-background: $primary-color;
./decidim-admin/app/packs/stylesheets/decidim/admin/utils/_settings.scss:804:$tab-color: $primary-color;
./decidim-admin/app/packs/stylesheets/decidim/admin/utils/_settings.scss:805:$tab-active-color: $primary-color;
./decidim-admin/app/packs/stylesheets/decidim/admin/utils/_settings.scss:820:$thumbnail-shadow-hover: 0 0 0 1px $primary-color;
./decidim-admin/app/packs/stylesheets/decidim/admin/utils/_mixins.scss:129:    background-color: $primary-color;
./decidim-admin/app/packs/stylesheets/decidim/admin/modules/_callouts.scss:4:  background-color: $primary-color;
./decidim-admin/app/packs/stylesheets/decidim/admin/modules/_filters.scss:45:      background-color: $primary-color;
./decidim-admin/app/packs/stylesheets/decidim/admin/modules/_table-list.scss:67:      color: $primary-color;
./decidim-admin/app/packs/stylesheets/decidim/admin/modules/_table-list.scss:121:      color: $primary-color;
./decidim-admin/app/packs/stylesheets/decidim/admin/_variables.scss:2:$primary-color: null;
./decidim_app-design/app/packs/stylesheets/public.scss:3:// $primary: map-get($foundation-palette, primary);
./decidim_app-design/app/packs/stylesheets/components/consultations/_consultations-header.scss:72:    color: $primary;
./decidim_app-design/app/packs/stylesheets/components/consultations/_consultations-banner.scss:7:    color: $primary;
./decidim_app-design/app/packs/stylesheets/components/banners/_top-banner.scss:2:  background-color: $primary;
./decidim_app-design/app/packs/stylesheets/components/evote/_evote.scss:217:    color: $primary;
./decidim_app-design/app/packs/stylesheets/components/evote/_evote.scss:237:    background: $primary;
./decidim-core/app/packs/stylesheets/decidim/_properties.scss:4:  --primary: #{$primary};
./decidim-core/app/packs/stylesheets/decidim/_properties.scss:5:  --primary-rgb: #{$primary-rgb};
./decidim-core/app/packs/stylesheets/decidim/utils/_settings.scss:72:  primary: $primary,
./decidim-core/app/packs/stylesheets/decidim/utils/_settings.scss:261:$accordion-background: $primary-color;
./decidim-core/app/packs/stylesheets/decidim/utils/_settings.scss:279:$accordionmenu-arrow-color: $primary-color;
./decidim-core/app/packs/stylesheets/decidim/utils/_settings.scss:291:$badge-background: $primary-color;
./decidim-core/app/packs/stylesheets/decidim/utils/_settings.scss:304:$breadcrumbs-item-color: $primary-color;
./decidim-core/app/packs/stylesheets/decidim/utils/_settings.scss:319:$button-background: $primary-color;
./decidim-core/app/packs/stylesheets/decidim/utils/_settings.scss:411:$drilldown-arrow-color: $primary-color;
./decidim-core/app/packs/stylesheets/decidim/utils/_settings.scss:501:$label-background: $primary-color;
./decidim-core/app/packs/stylesheets/decidim/utils/_settings.scss:601:$progress-meter-background: $primary-color;
./decidim-core/app/packs/stylesheets/decidim/utils/_settings.scss:684:$prototype-separator-background: $primary-color;
./decidim-core/app/packs/stylesheets/decidim/utils/_settings.scss:761:$slider-handle-background: $primary-color;
./decidim-core/app/packs/stylesheets/decidim/utils/_settings.scss:816:$tab-color: $primary-color;
./decidim-core/app/packs/stylesheets/decidim/utils/_settings.scss:817:$tab-active-color: $primary-color;
./decidim-core/app/packs/stylesheets/decidim/utils/_settings.scss:832:$thumbnail-shadow-hover: 0 0 0 1px $primary-color;
./decidim-core/app/packs/stylesheets/decidim/utils/_mixins.scss:169:    background-color: $primary;
./decidim-core/app/packs/stylesheets/decidim/email.scss:1193:  background: $primary;
./decidim-core/app/packs/stylesheets/decidim/email.scss:1194:  border: 2px solid $primary;
./decidim-core/app/packs/stylesheets/decidim/email.scss:1204:  border: 0 solid $primary;
./decidim-core/app/packs/stylesheets/decidim/email.scss:1293:  background: $primary;
./decidim-core/app/packs/stylesheets/decidim/modules/_buttons.scss:264:    primary: shade($primary, 50%),
./decidim-core/app/packs/stylesheets/decidim/modules/_space-stats.scss:26:      border-left: 2px solid $primary;
./decidim-core/app/packs/stylesheets/decidim/modules/_datepicker.scss:4:$calendar-active-color: $primary-color;
./decidim-core/app/packs/stylesheets/decidim/modules/_navbar.scss:5:$navbar-bg-focus: $primary;
./decidim-core/app/packs/stylesheets/decidim/modules/_statistics.scss:26:      border-left: 2px solid $primary;
./decidim-core/app/packs/stylesheets/decidim/_variables.scss:2:$primary-color: null;
./decidim-core/app/packs/stylesheets/decidim/_variables.scss:12:$primary: #cb3c29 !default;
./decidim-core/app/packs/stylesheets/decidim/_variables.scss:13:$primary-rgb: 239, 96, 77 !default;
./decidim-elections/app/packs/stylesheets/decidim/elections/focus/_evote.scss:223:    color: $primary;
./decidim-elections/app/packs/stylesheets/decidim/elections/focus/_evote.scss:245:    background: $primary;
./decidim-system/app/packs/stylesheets/decidim/system/application.scss:1:$primary-color: null;
```

alecslupu commented:
Short update: I am reviewing #10977 , and i am just wondering if the links on the cards should, or should not be Magenta ... 
![image](https://github.com/decidim/decidim/assets/105683/1604ef66-9fa7-4996-b54d-821edce42d6d)
 
Checking the code, i see that _cards.scss we have a definition like: 
```
.card__link {
  color: $anchor-color;
```
which points to _settings.scss where `$anchor-color: $secondary-color;`, end color being: #39747f, yet my organization colors are : 
![image](https://github.com/decidim/decidim/assets/105683/e5669ec2-ad35-42d7-ba20-6fb42ea5f3fa)

andreslucena commented:
@alecslupu I'm on the colors' rabbit hole, reviewing all the variables that you're mentioning. I'm also exploring using the spellchecker script, so we don't reintroduce these kinds of errors. I'll mark this PR as draft as it's going to be a bigger one, and I'll also ask for a review from @decidim/product just to be sure that we all agree with the new colors changes. 

I'll let you know when this is ready to be reviewed again.

andreslucena committed:
Use CSS variable for the autocomplete list

andreslucena committed:
Override the CSS variable for the autocomplete on the admin panel

andreslucena committed:
Use CSS variable for the assemblies' organization chart

andreslucena committed:
Use CSS variable for buttons

andreslucena committed:
Replace the 'anchor-color' SCSS variable with the 'secondary' CSS variable

andreslucena committed:
Add SCSS/CSS variables rules to the spellchecker

The idea is to stop regressions on this fix, by not letting developers
to commit changes to introduce these variables to the SCSS code:

* $primary
* $secondary
* $anchor-color

As they were the problematic. The only exceptions are the places were
the organization colors are not implemented:

* Admin panel
* System panel
* SCSS variables defintion files
* Emails (*)

(*) I think that on emails we should honor the organizations colors, but
this should be handled on other PR.

andreslucena committed:
Replace the 'secondary' SCSS variable with the 'secondary' CSS variable

andreslucena committed:
Replace the 'primary' SCSS variable with the 'primary' CSS variable

andreslucena committed:
Make the text-* and bg-* classes respect the organizations' colors

andreslucena committed:
Prevent leaking the CSS secondary variable to the admin panel

andreslucena commented:
This is ready for a first review. @carolromero I'll assign it to you for checking this one as it's in a big part changing the current design when changing the organizations' colors.

andreslucena committed:
Fix stylelint offenses

andreslucena committed:
Merge branch 'develop' into fix/modal-primary-colors

carolromero commented:
@andreslucena I found a couple of places where it would need to be changed:

* The colour for the follow button is the secondary (and hollow type):
![image](https://github.com/decidim/decidim/assets/23509895/1e235ce3-246a-4270-b1e6-70393437cdc0)
* The chevron of the sorting filters needs to be changed to the secondary colour.
![image](https://github.com/decidim/decidim/assets/23509895/a8b416b5-ccaa-4e47-992e-1f03ca53ef7f)

It _really_ is a rabbit hole 🐰. I see that the OSP team has tried to resolve this inconsistency, you might want to check the selectors in this instance in case we have missed any: https://participez.nanterre.fr/processes/ZAC-Chemindelile

> Short update: I am reviewing https://github.com/decidim/decidim/pull/10977 , and i am just wondering if the links on the cards should, or should not be Magenta ...

@alecslupu I've been reviewing the larger instances (that are not too heavily tuned), and many of them, consciously or not, have chosen not to touch the secondary colour. So keeping the links with the secondary colour would be the least disruptive change when applying the fix.

andreslucena commented:
> The colour for the follow button is the secondary (and hollow type):

The problem here is that the follow button is the redesigned one. As we're in the middle of merging that and this is going to change (a lot) and it may not make sense anymore for v0.28, I think its better to leave this as a draft and wait for the merge and stabilization of develop with the new code to happen. 

After that, we can go back to this one to see better if this problem is still happening in develop and which is the best strategy to follow. For the moment, there are still the SASS variables that generate the problem in the `feature/redesign` branch (ie `$primary`, `$secondary`) but I don't know if that's the final form.

andreslucena commented:
After #11685 this looks like I was expecting: lots of conflicts, as my fix was based on v0.27 styling. 

By searching the culprits of this issue, the SCSS variables (`$primary`, `$secondary` and `$anchor-color`), that were used instead of the CSS variables (`var(--primary)` and `var(--secondary)`), I can say that **this issue isn't happening in v0.28.0.dev anymore**. Also, as we made these colours random in the seeds (#10974), these kinds of problems are much easier to detect.

Even though v0.27.x is still maintained, I personally want to focus in the release of the next v0.28.0 release, and I have doubts on introducing a change like this in a patch release. This will change colours of the installations, and most likely will break the current designs.

Saying all that, I'll close this PR, at least for now. I'll also comment it in the next @decidim/product meeting, just in case we want to recover this approach. In that case, we would open a new PR as we will change the base branch to `release/0.27-stable`.
